### PR TITLE
Beta

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #180, #206 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [x] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Enhance the component with theme support using design tokens, fix color theming issues, and refactor component registration for better performance. Update dependencies and documentation to reflect these changes, and remove outdated scripts.

New Features:
- Introduce theme support for the component, allowing it to be restyled using design tokens.

Bug Fixes:
- Resolve minor color theming issues to ensure consistent appearance across components.

Enhancements:
- Refactor custom component registration to use a new utility for versioned component tags.
- Update dependencies and file extensions to improve performance and maintainability.

Build:
- Add a new ESLint configuration file to standardize code quality checks.

Documentation:
- Update documentation to include new theme support and usage examples with updated design tokens.

Chores:
- Remove obsolete scripts and update changelog with recent changes.

### CODE REVIEW

File: demo/api.md [hunk 1]
```diff
@@ -1033,6 +1032,11 @@ export function populateSlotContentExample() {
         slotElement.setAttribute('slot', slotName);
         slotElement.textContent = item.content;
 
+        // Set the 'highlight' attribute on date slot content
+        if (item.slot === 'date' && item.highlight) {
+          slotElement.setAttribute('highlight', item.highlight);
+        }
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply demo/api.md:1038]: Not adding documentation at this time.

File: demo/api.md [hunk 2]
```diff
@@ -1082,4 +1086,40 @@ export function alertValueExample() {
 }
 ```
 <!-- AURO-GENERATED-CONTENT:END -->
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply demo/api.md:1088]: Not adding short descriptions at this time.

**Review by jason-capsule42 — APPROVED** [review]

**Pull request merged**